### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by tomdan-ai**
# In This Milestone
### I implemented the following:

- Add validations for the following:
  - The User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - The Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all my models:
  - `user.rb`
  - `post.rb`
  - `comment.rb`
  - `like.rb`
- As found in the following files:
  - `user_spec.rb`
  - `post_spec.rb`
  - `comment_spec.rb`
  - `like_spec.rb`

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  end
+  context 'Validating Users Data' do
+    subject(:user) { User.new(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland') }
+    before { user.save }
+
+    it 'Name of user ahould be valid' do
+      expect(user).to be_valid
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/models/user_spec.rb]: - [ ] Kindly note that we are also required to test custom model methods. Kindly add tests for the set_defaults and recent posts methods in the User model.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    it 'should have likes_counter equal to 0' do
+      expect(subject.likes_counter).to eq(0)
+    end
+
+    it 'title should not be greater than 250 characters' do
+      post = Post.create(author: user, title: 'a' * 251, text: 'My First Post')
+      expect(post).not_to be_valid
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/models/post_spec.rb]: - [ ] Kindly add tests for the set_defaults, increment counter and recent comments methods in the Post model.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @tomdan-ai,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @tomdan-ai, 




Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡
 
✅ Descriptive PR 👍 
✅ Nice implementing the changes 🔥 
✅ Linters are passing 👏 


## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**